src/OrchardCore.Modules/OrchardCore.ArchiveLater/Services/ScheduledArchivingBackgroundTask.cs

```diff
@@ -0,0 +1,50 @@
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using OrchardCore.ArchiveLater.Indexes;
+using OrchardCore.BackgroundTasks;
+using OrchardCore.ContentManagement;
+using OrchardCore.Modules;
+using YesSql;
+
+namespace OrchardCore.ArchiveLater.Services;
+
+[BackgroundTask(Schedule = "* * * * *", Description = "Archives content items when their scheduled archive date time arrives.")]
+public class ScheduledArchivingBackgroundTask : IBackgroundTask
+{
+    private readonly ILogger<ScheduledArchivingBackgroundTask> _logger;
+    private readonly IClock _clock;
+
+    public ScheduledArchivingBackgroundTask(ILogger<ScheduledArchivingBackgroundTask> logger, IClock clock)
+    {
+        _logger = logger;
+        _clock = clock;
+    }
+
+    public async Task DoWorkAsync(IServiceProvider serviceProvider, CancellationToken cancellationToken)
+    {
+        var itemsToArchive = await serviceProvider
+            .GetRequiredService<ISession>()
+            .QueryIndex<ArchiveLaterPartIndex>(index => index.Latest && index.Published && index.ScheduledArchiveDateTimeUtc < _clock.UtcNow)
+            .ListAsync();
+
+        if (!itemsToArchive.Any())
+        {
+            return;
+        }
+
+        var contentManager = serviceProvider.GetRequiredService<IContentManager>();
+
+        foreach (var item in itemsToArchive)
+        {
+            var contentItem = await contentManager.GetAsync(item.ContentItemId);
```

Comment:
So here we could retrieve the part and then (maybe after a null check on the part) set to null here the schedule date